src/Persistence/PaYaml/Serialization/YamlSequenceTesterConverter.cs

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Collections;
+using YamlDotNet.Core;
+using YamlDotNet.Core.Events;
+using YamlDotNet.Serialization;
+
+namespace Microsoft.PowerPlatform.PowerApps.Persistence.PaYaml.Serialization;
+
+/// <summary>
+/// custom converter used to test if the input is a sequence of YAML items.
+/// </summary>
+internal sealed class YamlSequenceTesterConverter : IYamlTypeConverter
+{
+    public bool Accepts(Type type)
+    {
+        return type == typeof(IEnumerable) || type.IsSubclassOf(typeof(IEnumerable)) ||
+            type == typeof(IEnumerable<object>) || type.IsSubclassOf(typeof(IEnumerable<object>)) ||
+            type == typeof(object[]);
+    }
+
+    public object? ReadYaml(IParser parser, Type type)
+    {
+        if (parser.Current is not SequenceStart)
```

Comment:
when using the `SequenceDetectorResult` pattern I mentioned above, this converter could just have this SequenceStart check, completely skip the contents, and then return the result value.

the benefit here is that there wouldn't be an exception thrown for code flow control.